book/src/todo_app_1.md

```diff
@@ -0,0 +1,231 @@
+
+
+<div style="text-align:center"><img src="img/todo_row.png" /></div>
+
+Again, we describe the mockup with a composite template.
+
+
+<span class="filename">Filename: listings/todo_app/1/todo_row/todo_row.ui</span>
+
+```xml
+{{#rustdoc_include ../listings/todo_app/1/todo_row/todo_row.ui}}
+```
+
+First we [derive](https://docs.gtk.org/gtk4/class.Box.html#hierarchy) `TodoRow` from `gtk:Box`.
+
+<span class="filename">Filename: listings/todo_app/1/todo_row/mod.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/todo_row/mod.rs:glib_wrapper}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+In `imp::TodoRow`, we hold references to `completed_button` and `content_label`.
+We also store a mutable vector of bindings.
+Why we need that will become clear as soon as we get to bind the state of `TodoObject` to the corresponding `TodoRow`.
+
+
+<span class="filename">Filename: listings/todo_app/1/todo_row/imp.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/todo_row/imp.rs:struct_and_subclass}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+Now we can bring everything together.
+We override the `imp::Window::constructed` in order to ensure that everything will be setup immediately after the parent is constructed.
+
+<span class="filename">Filename: listings/todo_app/1/window/imp.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/window/imp.rs:constructed}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+Since we need to access the model quite often, we add the convenience method `Window::model` for that.
+In `Window::setup_model` we create a new model.
+Then we store a reference to the model in `imp::Window` as well as in `gtk::ListView`.
+
+<span class="filename">Filename: listings/todo_app/1/window/mod.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/window/mod.rs:model}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+In `Window::setup_callbacks` we connect to the "activate" signal of the entry.
+Whenever we finished writing up a new task, we can add it by activating the entry.
+Pressing the enter key is one way to activate the entry.
+Then, a new `TodoObject` with the content will be created and appended to the model.
+Finally, the entry will be cleared.
+
+<span class="filename">Filename: listings/todo_app/1/window/mod.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/window/mod.rs:setup_callbacks}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+Again the list elements for the `gtk::ListView` are produced by a factory.
+Before we move on to the implementation, let us take a step back and think about which behavior we expect here.
+`content_label` of `TodoRow` should follow `content` of `TodoObject`.
```

Comment:
```suggestion
`content_label` of `TodoRow` should be equal to `content` of `TodoObject`.
```

Reply:
I think "follow" describes it better that this relationship is one-sided.